src/syngen/ml/utils/utils.py

```diff
@@ -130,7 +130,15 @@ def fetch_dataset(dataset_pickle_path: str):
     Deserialize and return the object of class Dataset
     """
     with open(dataset_pickle_path, "rb") as f:
-        return pickle.loads(f.read())
+        dataset = pickle.loads(f.read())
+    if hasattr(dataset, "df"):
```

Comment:
Add comment explaining the logic, like,
Check if the serialized class has associated dataframe and drop it as it might contain sensitive data. Save columns from the dataframe for later use.

Reply:
Thank you for your comment) I'll add this info in the comment